share/wake/lib/versions/v0_22.wake

```diff
@@ -1,7 +1,108 @@
+from v0_24_wake export def unary ! + +. - -. ~ ∏ ∏. ∑ ∑. √ ∛ ∜
+from v0_24_wake export def binary != !=* !=. !=^ !=~ $ % && * *. + ++ +. , - -. . / /. /../ // /| ; < <* <. << <= <=* <=. <=> <=>* <=>. <=>^ <=>~ <=^ <=~ <^ <~ == ==* ==. ==/ ==^ ==~ > >* >. >= >=* >=. >=^ >=~ >> >^ >~ ^ ^. | || → ∈ ∉ ∋ ∌ ∘ ∩ ∪ ⊆ ⊇ ⊈ ⊉ ⊊ ⊋ ⊎
+from v0_24_wake export topic environment path runner source
+
+from v0_24_wake export def getJobResultOutput=getJobOutput getJobResultOutputs=getJobOutputs
+
+
+# Functions returning a single `Path` can easily extract it from the `Result`
+
+def wrapBadPath (pathResult: Result Path Error): Path =
+    match pathResult
+        Pass path  = path
+        Fail error = makeBadPath error
+
+
+export def claimFileAsPath (existingFile: String) (desiredWorkspacePath: String): Path =
+    from v0_24_wake import claimFileAsPath
+    claimFileAsPath existingFile desiredWorkspacePath
+    | wrapBadPath
+
+export def claimFileAsPathIn (outputDirectory: Path) (existingFile: String) (desiredWorkspacePath: String): Path =
+    from v0_24_wake import claimFileAsPathIn
+    claimFileAsPathIn outputDirectory existingFile desiredWorkspacePath
+    | wrapBadPath
+
+export def getJobOutput (job: Job): Path =
+    from v0_24_wake import getJobOutput
+    getJobOutput job
+    | wrapBadPath
+
+export def installAs (dest: String) (file: Path): Path =
+    from v0_24_wake import installAs
+    installAs dest file
+    | wrapBadPath
+
+export def installIn (toRoot: String) (fromRoot: String) (sourcePath: Path): Path =
+    from v0_24_wake import installIn
+    installIn toRoot fromRoot sourcePath
+    | wrapBadPath
+
+export def mkdir (path: String): Path =
+    from v0_24_wake import mkdir
+    mkdir path
+    | wrapBadPath
+
+export def mkdirIn (parent: Path) (mode: Integer) (name: String): Path =
+    from v0_24_wake import mkdirIn
+    mkdirIn parent mode name
+    | wrapBadPath
+
+export def write (path: String) (content: String): Path =
+    from v0_24_wake import write
+    write path content
+    | wrapBadPath
+
+export def writeIn (parent: Path) (mode: Integer) (name: String) (content: String): Path =
+    from v0_24_wake import writeIn
+    writeIn parent mode name content
+    | wrapBadPath
+
+export def source (file: String): Path =
+    from v0_24_wake import source
+    source file
+    | wrapBadPath
+
+
+# Every other function returning `Result` post-0.24 needs some degree of reimplementation
+
+def treeCompat job typ =
+    def tree j t =
+        prim "job_tree"
+    def treeOk (Pair f h) = match h
+        "BadHash" = makeBadPath (makeError "Could not hash {f}")
+        _ = claimFileAsPath f "build/v0_22_wake/{f}"
```

Comment:
I found something else that I wanted to ask about: I don't like this `claimFileAsPath`, but is there any better way of doing it?  Since we obviously don't want to export the `Path` constructor, I can't just wrap the valid path names as in job.wake.  `source` is also out since it's likely returning files that were created by the job rather than ones in Git, and I can't just base it off the `Result (List Path) Error` that the stdlib now returns as the `Fail` case for that throws out all the good paths.

Reply:
Oh jeeze. Yeah, I overlooked that.

Reply:
So, yeah, we also don't really want to export makeBadPath in v0_24_wake, either.

Reply:
Here's my proposal: move the `data Path` fully exported into a new package `internals_wake`. Then, re-export only the type from `wake`. Then, you can define `makeBadPath` in `v0_22_wake` and also construct a `Path` directly there.

Reply:
That was what I came up with as well.  It seemed ugly, but if you don't have anything better, I guess it's what we'll need to be doing.

Reply:
The other option is to use a different `Path` type in `v0_22` and `v0_24`, and convert between them as `Path@v0_22_wake => Result Path@v0_24_wake Error`. However, I think that will be more painful to support upgrading extant code bases.

Reply:
If they were mostly staying within file boundaries, I can see that *maybe* working, but `Path`s get passed around way too often for me to think that would work.

Reply:
Looks good.